static/js/stream_edit.js

```diff
@@ -576,6 +576,34 @@ $(function () {
         e.stopPropagation();
     });
 
+    $("#subscriptions_table").on("click", ".deactivate", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        var stream_name = stream_data.maybe_get_stream_name(get_stream_id(e.target));
+        var stream_id = get_stream_id(e.target);
+        var deactivate_stream_modal = templates.render("deactivation-stream-modal");
+        $(".subscription_settings").append(deactivate_stream_modal);
+        $("#deactivation_stream_modal .stream_name").text(stream_name);
```

Comment:
This should be its own commit on top, but I think we should make `stream_name` a template variable for the modal, rather than patching it in JS.  In general, we try to avoid patching content in JS when it can be done in the template, since it's messier and harder to read the code that way.

Reply:
I have tried this but it results in more code and complexity. 
This is called in organization setting, where templates are added to `#settings_overlay_container` on overlay load event. If we add template variable then we have to render that template on time we get click on any specific stream, which also means we have to manually delete the DOM element when deactivated modal is dismissed or submit.
Is there any other way?